pkg/hyper/client.go

```diff
@@ -189,6 +189,48 @@ func (c *Client) GetContainerInfo(container string) (*types.ContainerInfo, error
 	return cinfo.ContainerInfo, nil
 }
 
+// StopContainer stops a hyper container
+func (c *Client) StopContainer(containerID string, timeout int64) error {
+	//do checks about container and pod status
+	containerInfo, err := c.GetContainerInfo(containerID)
+	if err != nil {
+		return err
+	}
+
+	podInfo, err := c.GetPodInfo(containerInfo.PodID)
+	if err != nil {
+		return err
+	}
+
+	if podInfo.Status.Phase != "running" && podInfo.Status.Phase != "Running" {
+		return fmt.Errorf("Pod %s is not running.", containerInfo.PodID)
+	}
+
+	if containerInfo.Status.Phase != "running" {
+		return fmt.Errorf("Container %s is not running.", containerID)
+	}
+
+	ch := make(chan error, 1)
+
+	go func(containerID string) {
+		ch <- nil
```

Comment:
Code will be blocked util we get something [here](https://github.com/kubernetes/frakti/pull/22/files#diff-f7ec4be94a67a866942db2ebd145d763R224).
So, the timeout will be more precise.